airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml

```diff
@@ -94,6 +98,45 @@ definitions:
       message:
         description: "the log message"
         type: string
+  AirbyteTraceMessage:
+    type: object
+    additionalProperties: true
+    required:
+      - type
+      - emitted_at
+    properties:
+      type:
+        description: "the type of trace message"
+        type: string
+        enum:
+          - ERROR
+      emitted_at:
+        description: "the time in ms that the message was emitted"
+        type: number
+      error:
+        description: "error data from the sync"
+        "$ref": "#/definitions/AirbyteErrorTraceMessage"
+  AirbyteErrorTraceMessage:
+    type: object
+    additionalProperties: true
+    required:
+      - message
+    properties:
+      message:
+        description: The human readable message that will be shown to the user in the UI
+        type: string
+      internal_message:
+        description: The less human readable, more technical message showing the reason for the failure, e.g. the first line of a stack trace
+        type: string
+      stack_trace:
+        description: The full stack trace of the error
+        type: string
+      failure_type:
+        description: The type of error
+        type: string
+        enum:
+          - SYSTEM_ERROR
+          - CONFIG_ERROR
```

Comment:
Noting that these enum values and https://github.com/airbytehq/airbyte/blob/master/airbyte-config/models/src/main/resources/types/FailureReason.yaml#L20-L26 have different cases. Not sure if that matters

Reply:
Since we will need to convert between the POJO type generated from this json schema to the type generated by the FailureReason json schema, it should be pretty trivial to lowercase these values when performing the conversion.

However, given that there are already other lowercase string enum values in this protocol file (e.g. [SyncMode](https://github.com/airbytehq/airbyte/blob/124a3c6cb9091e83ca8da047eae88b5d8aebeb9a/airbyte-protocol/models/src/main/resources/airbyte_protocol/airbyte_protocol.yaml#L201-L203)), we might as well make these enum values lowercase as well to have them be consistent with the FailureReason values.